### ISSUE TITLE
Investigate error handling when logger_path becomes invalid and logger_plugin is filesystem

### ISSUE THREAD

**Issue by theopolis** [facebook/osquery#2120]
When starting the daemon I know an error message is reported to stdout, but the daemon is generally useless when the `logger_path` is invalid when `logger_plugin=filesystem`. The daemon will exit `78` if this logging configuration is unusable.

What about when the logging configuration "becomes" unusable during the execution of the daemon?

**Comment by theopolis** [comment]
Seems like the current implementation of the filesystem logger plugin is to exit 78 when the results log file cannot be created/used.
```
./build/debug_darwin/osquery/osqueryd --verbose --ephemeral --disable_database                   
I0527 08:29:47.167816 2029744128 init.cpp:318] osquery initialized [version=1.7.4-29-g10664a4-debug]
I0527 08:29:47.168889 2029744128 extensions.cpp:199] Could not autoload extensions: Failed reading: /etc/osquery/extensions.load
I0527 08:29:47.168961 2029744128 dispatcher.cpp:87] Adding new service: 0x7fe4b1e1ed88 to thread: 0x7fe4b1e1ebb8
I0527 08:29:47.168987 2029744128 dispatcher.cpp:96] Thread: 0x7fff78fb7000 requesting a join
I0527 08:29:47.170202 528384 watcher.cpp:365] osqueryd watcher (60000) executing worker (60001)
I0527 08:29:47.191329 2029744128 init.cpp:315] osquery worker initialized [watcher=60000]
I0527 08:29:47.192201 2029744128 dispatcher.cpp:87] Adding new service: 0x7fabf8f000d8 to thread: 0x7fabf8f001d8
I0527 08:29:47.192296 2029744128 extensions.cpp:206] Could not autoload modules: Failed reading: /etc/osquery/modules.load
W0527 08:29:47.192713 2029744128 init.cpp:556] Error reading config: config file does not exist: /var/osquery/osquery.conf
E0527 08:29:47.192795 2029744128 init.cpp:492] Cannot activate filesystem logger plugin: Could not create file: /var/log/osquery/osqueryd.results.log
[...]
~/git/github/osquery (branch:master) [78]»  
```

**Comment by theopolis** [comment]
However, if while the daemon is running, I remove the logger path (directory):
```
Could not create log file: No such file or directory
COULD NOT CREATE LOGFILE '20160527-083733.60736'!
Could not create log file: No such file or directory
COULD NOT CREATE LOGFILE '20160527-083733.60736'!
E0527 08:37:33.674084 3825664 scheduler.cpp:117] Error logging the results of query ([...]): Could not create file: /tmp/TEST_DIR/osqueryd.results.log
```

Repeats for each query attempted, and the daemon does not exit.

**Comment by theopolis** [comment]
I think a very basic solution here is to call `Initializer::requestShutdown()` similar to: https://github.com/facebook/osquery/blob/master/osquery/dispatcher/scheduler.cpp#L111

The second argument is a string line to be printed to the system log before exiting. We can add something helpful like, `Cannot write logs to file: path`

**Comment by vjon** [comment]
This was fixed by #2700, but didn't automatically close for some reason.
